src/clients/authorization/ResourceDetails.tsx

```diff
@@ -0,0 +1,393 @@
+import { ResourceDetailsParams, toResourceDetails } from "../routes/Resource";
+import { KeycloakSpinner } from "../../components/keycloak-spinner/KeycloakSpinner";
+import { useAdminClient, useFetch } from "../../context/auth/AdminClient";
+import { HelpItem } from "../../components/help-enabler/HelpItem";
+import { ViewHeader } from "../../components/view-header/ViewHeader";
+import { useConfirmDialog } from "../../components/confirm-dialog/ConfirmDialog";
+import { useAlerts } from "../../components/alert/Alerts";
+import { FormAccess } from "../../components/form-access/FormAccess";
+import {
+  convertToMultiline,
+  MultiLine,
+  MultiLineInput,
+  toValue,
+} from "../../components/multi-line-input/MultiLineInput";
+import { toClient } from "../routes/Client";
+import { ScopePicker } from "./ScopePicker";
+import {
+  arrayToAttributes,
+  attributesToArray,
+  KeyValueType,
+} from "../../components/attribute-form/attribute-convert";
+import { AttributeInput } from "../../components/attribute-input/AttributeInput";
+
+import "./resource-details.css";
+
+type FetchResource = {
+  client?: ClientRepresentation;
+  resource?: ResourceRepresentation;
+  permissions?: ResourceServerRepresentation[];
+};
+
+type SubmittedResource = Omit<ResourceRepresentation, "attributes" | "uris"> & {
+  attributes: KeyValueType[];
+  uris: MultiLine[];
+};
+
+export default function ResourceDetails() {
+  const { t } = useTranslation("clients");
+  const [client, setClient] = useState<ClientRepresentation>();
+  const [resource, setResource] = useState<ResourceRepresentation>();
+
+  const [permissions, setPermission] =
+    useState<ResourceServerRepresentation[]>();
+
+  const adminClient = useAdminClient();
+  const { addAlert, addError } = useAlerts();
+  const form = useForm<SubmittedResource>({
+    shouldUnregister: false,
+    mode: "onChange",
+  });
+  const { register, errors, control, setValue, handleSubmit } = form;
+
+  const { id, resourceId, realm } = useParams<ResourceDetailsParams>();
+  const history = useHistory();
+
+  const setupForm = (resource?: ResourceRepresentation) => {
+    Object.entries(resource || {}).map(([key, value]) => {
+      if (key === "uris") {
+        setValue("uris", convertToMultiline(value));
+      } else if (key === "attributes") {
+        setValue("attributes", attributesToArray(value));
+      } else {
+        setValue(key, value);
+      }
+    });
+  };
+
+  useFetch(
+    async (): Promise<FetchResource> => {
+      const client = await adminClient.clients.findOne({ id });
+      if (resourceId) {
+        const [resource, permissions] = await Promise.all([
+          adminClient.clients.getResource({
+            id,
+            resourceId,
+          }),
+          adminClient.clients.listPermissionsByResource({
+            id,
+            resourceId,
+          }),
+        ]);
+
+        return { client, resource, permissions };
+      }
+      return { client };
+    },
```

Comment:
Since the other API calls have no dependency on `client` it's call can also be done in parallel.
```suggestion
    async (): Promise<FetchResource> => {
      const [client, resource, permissions] = await Promise.all([
        adminClient.clients.findOne({ id }),
        resourceId
          ? adminClient.clients.getResource({ id, resourceId })
          : Promise.resolve(undefined),
        resourceId
          ? adminClient.clients.listPermissionsByResource({ id, resourceId })
          : Promise.resolve(undefined),
      ]);

      return { client, resource, permissions };
    },
```